src/lib.rs

```diff
@@ -345,11 +347,17 @@ pub enum SynthesisError {
 
 /// Represents a constraint system which can have new variables
 /// allocated and constrains between them formed.
-pub trait ConstraintSystem<E: ScalarEngine>: Sized {
+pub trait ConstraintSystem<E: ScalarEngine>: Sized + Send {
     /// Represents the type of the "root" of this constraint system
     /// so that nested namespaces can minimize indirection.
     type Root: ConstraintSystem<E>;
 
+    fn new() -> Self {
+        unimplemented!(
```

Comment:
`new` without arguments is a rust antipattern, should be `default`, aka just requiring `Default` as an additional trait bound

Reply:
I was hoping you wouldn't say that…

Reply:
And now I remember why: there are problems trying to make that work with all extant `ConstraintSystem` implementations. If you can make that compile just within this repo, please make a PR to this one. (I will be happy, but I will also accept the status quo presented here if it becomes a matter of extreme contortionism. If a different name would be less annoying for you, I'm open to suggestions.)